### PR TITLE
WSTEAMA-325 - Simorgh routes /kyrgyz to the HomePage component

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves [WSTEAMA 325](https://jira.dev.bbc.co.uk/browse/WSTEAMA-325)

Overall changes
======
Implements logic to ensure that `/kyrgyz` no longer matches the FrontPage route, but instead matches the HomePage route instead, on local and test environments only. 

Code changes
======

- Adds unit tests which check valid routes on local and test environments as we migrate FrontPage routes to HomePage.
- Renames instances of HomePagePath which use the `/tipohome` route to TipoHomePath
- Adds logic to ensure that `/kyrgyz` matches the HomePage route instead of FrontPage, on local and test.

Testing
======
1. _List the steps used to test this PR._

Helpful Links
======

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/routes/utils/regex/index.js [hunk 1]
```diff
@@ -42,12 +41,12 @@ export const frontPageDataPath = `${frontPagePath}.json`;
 export const frontPageSwPath = getSwRegex(allServices);
 export const frontPageManifestPath = getManifestRegex(allServices);
 
+export const tipoHomePath = getTipoHomeRegex(allServices);
+export const tipoHomeDataPath = `${tipoHomePath}.json`;
+
 export const homePagePath = getHomePageRegex(allServices);
 export const homePageDataPath = `${homePagePath}.json`;
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/index.js:48]: Reminder to remove this as I don't think we need it anymore

Comment (karinathomasbbc) [reply src/app/routes/utils/regex/index.js:48]: Nope, we need it!

File: src/app/routes/utils/regex/utils/index.js [hunk 2]
```diff
@@ -31,15 +33,28 @@ export const getArticleManifestRegex = services => {
...
   const serviceRegex = getServiceRegex(services);
   return `/:service(${serviceRegex}):variant(${variantRegex})?/tipohome:amp(${ampRegex})?`;
 };
 
+export const getHomePageRegex = () => {
+  if (!isLive()) {
+    const homePageServiceRegex = getServiceRegex(['kyrgyz']);
+    return `/:service(${homePageServiceRegex}):variant(${variantRegex})?:amp(${ampRegex})?`;
+  }
+  return '';
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/utils/index.js]: Let's try to remove this and see what happens? 🤔

File: src/app/routes/utils/regex/index.test.js [hunk 3]
```diff
@@ -555,3 +532,54 @@ describe('legacyAssetPageDataPath', () => {
   });
   shouldNotMatchInvalidRoutes(invalidDataRoutes, legacyAssetPageDataPath);
 });
+
+describe('frontPage -> homePage migration', () => {
+  const services = Object.keys(serviceConfig);
+
+  const homePageServices = ['kyrgyz'];
+  const homePageRoutes = homePageServices.map(service => `/${service}`);
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/index.test.js:537]: Let's also check the `homePageData` and `frontPageData` routes in these tests

```
/** Pseudocode:
Create another variable for the homePageDataRoutes, which appends .json to each of the entries in homePageRoutes

Check if the homePageDataRegex matches the valid homePageData routes in the `homePage regex on the %s environment` test below
**/
```

File: src/app/routes/utils/regex/utils/index.js (hunk 2 above)
Comment (louisearchibald) [reply src/app/routes/utils/regex/utils/index.js]: Removed

File: src/app/routes/utils/regex/index.test.js [hunk 4]
```diff
@@ -8,10 +8,11 @@ import {
...
   frontPageSwPath,
-  homePagePath,
-  homePageDataPath,
-  homePageManifestPath,
-  homePageSwPath,
+  tipoHomePath,
+  tipoHomeDataPath,
+  // add in tests in frontPage migrations tests
+  // homePagePath,
+  // homePageDataPath,
```
Comment (karinathomasbbc) [inline src/app/routes/utils/regex/index.test.js]: We can remove these comments now
```suggestion
```

File: src/app/routes/utils/regex/index.test.js (hunk 3 above)
Comment (karinathomasbbc) [reply src/app/routes/utils/regex/index.test.js:537]: Nope - this is no longer required!

File: src/app/routes/utils/regex/index.test.js (hunk 4 above)
Comment (louisearchibald) [reply src/app/routes/utils/regex/index.test.js]: Removed! Thanks 😄

**Review by mfonofm — APPROVED** [review]

**Review by amoore108 — APPROVED** [review]

**Pull request merged**